docs/source/howto/aforms.rst

```diff
@@ -0,0 +1,56 @@
+.. _aforms:
+
+How to extract AcroForm fields from a PDF using Python and PDFMiner
+********************************
+
+Before you start, make sure you have :ref:`installed pdfminer.six<install>`.
+
+The second thing you need is a PDF with AcroForms (as found in PDF files with fillable forms or multiple choices).
+
+It won't work for XFA forms.
+
+.. code-block:: python
+
+    from pdfminer.pdfparser import PDFParser
+    from pdfminer.pdfdocument import PDFDocument
+    from pdfminer.pdftypes import resolve1, PDFObjRef
+    from pdfminer.psparser import PSLiteral, PSKeyword
+    from pdfminer import utils
+    
+    data = {}
+    
+    with open(file_path, 'rb') as fp:
+        parser = PDFParser(fp)
+        
+        doc = PDFDocument(parser)
+        res = resolve1(doc.catalog)
+
+        if 'AcroForm' in res:
+            fields = resolve1(doc.catalog['AcroForm'])['Fields']
+
+            for f in fields:
+                field = resolve1(f)
+                name, value = field.get('T'), field.get('V')
+
+                # decode name
+                name = name.decode()
+
+                if value:
+
+                    # resolve indirect obj
+                    while isinstance(value, PDFObjRef):
+                        value = resolve1(value)
+
+                    # decode PSLiteral, PSKeyword
+                    if isinstance(value, (PSLiteral, PSKeyword)):
+                        value = value.name
+
+                    # decode bytes
+                    if isinstance(value, bytes):
+                        value = utils.decode_text(value)
```

Comment:
Not for this PR, but in general it seems silly to me that users have to hope with different types here to decode. Perhaps we should add a util function that does all these if statements for you? (cc @pietermarsman do you think that's worth an issue?)

Reply:
It could make sense. There are already some utility functions for this in _pdftypes.py_: `resolve1()`, `resolve_all()`, `decipher_all()` and methods for resolving specific types. 

Not sure if `resolve1` -> `get name` -> `decoding` is a common pattern though.

Reply:
I don't know if this is common, but my form needed it because I could get those types as a value... or combinations of them, like value as an indirect object which resolved into a byte encoded PSLiteral... no idea how one does that.

The key were better, they just needed proper decoding (being byte). 

Btw: what does decipher_all() do?